File: javascript/6-kyu/the-supermarket-queue/solution.js
function queueTime(customers, n) {
  let arr = new Array(n).fill(0);

  for (let i = 0; i < customers.length; i++) {
    let index = arr.indexOf(Math.min(...arr));
    arr[index] += customers[i];
  }

  return Math.max(...arr);
}

File: csharp/6-kyu/the-supermarket-queue/solution.cs
using System;
using System.Collections;
using System.Linq;
public class Kata
{
  public static long QueueTime(int[] customers, int n)
  {

    if (n <= 1)
    {
      return customers.Sum();
    }
    else if (customers.Max() < n)
    {
      return customers.Max();
    }
    else
    {
      Queue line = new Queue();
      for (var i = 0; i < customers.Length; i++)
      {
        line.Enqueue(customers[i]);
      }
      while (line.Count > 0)
      {

      }
      return (line);
    }
  }
}